src/lib/Microsoft.Fx.Portability.Reports.DGML/ReferenceNode.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Fx.Portability.Reporting.ObjectModel;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Fx.Portability.Reports.DGML
+{
+    class ReferenceNode
+    {
+        private bool _searchInGraph = false;
+
+        public List<TargetUsageInfo> UsageData { get; set; }
+
+        public string Assembly { get; set; }
+
+        public bool Unresolved { get; set; }
+
+        public HashSet<ReferenceNode> Nodes { get; set; }
+        public bool IsMissing { get; internal set; }
+
+        public string SimpleName
+        {
+            get
+            {
+                if (!IsMissing)
+                    return new AssemblyName(Assembly).Name;
+
+                return "Unresolved: " + new AssemblyName(Assembly).Name;
+            }
+        }
+
+        public ReferenceNode(string AssemblyName, bool unresolved = false)
+        {
+            Assembly = AssemblyName;
+            this.Unresolved = unresolved;
+            Nodes = new HashSet<ReferenceNode>();
```

Comment:
Do we also need to pass in your `ReferenceNodeComparer` for the HashSet? Might be better to have a `ReferenceNodeComparer.Instance` so we don't have to create multiple instances of this comparer.